### PR TITLE
Leave management application2

### PR DESCRIPTION
**Pull request by mukhil-geeky**

### CODE REVIEW

File: src/services/WolfDen.API/Controllers/LeaveManagement/LeaveRequestController.cs [hunk 1]
```diff
@@ -16,5 +23,48 @@ public async Task<LeaveRequestHistoryResponseDto> GetLeaveRequestHistory([FromQu
...
             return await _mediator.Send(query,cancellationToken); 
         }
+
+        [HttpPost]
+        public async Task<bool> ApplyLeaveRequest( [FromBody] AddLeaveRequestCommand command,CancellationToken cancellationToken)
+        {
+            return await _mediator.Send(command, cancellationToken);
+        }
+
+        [HttpPatch("revove-leave")]
```
Comment (zameelnp) [inline src/services/WolfDen.API/Controllers/LeaveManagement/LeaveRequestController.cs:33]: typo and pass id here

File: src/services/WolfDen.API/Controllers/LeaveManagement/LeaveRequestController.cs [hunk 2]
```diff
@@ -16,5 +23,48 @@ public async Task<LeaveRequestHistoryResponseDto> GetLeaveRequestHistory([FromQu
...
+        public async Task<bool> RevokeLeave([FromBody] RevokeLeaveRequestCommand command,CancellationToken cancellationToken)
+        {
+            return await _mediator.Send(command, cancellationToken);
+        }
+
+        [HttpGet("subordinate-leave-requets{id}/{status}")]
```
Comment (zameelnp) [inline src/services/WolfDen.API/Controllers/LeaveManagement/LeaveRequestController.cs:39]:
```suggestion
        [HttpGet("subordinate-leave-requests/{id}/{status}")]
```

File: src/services/WolfDen.API/Controllers/LeaveManagement/LeaveRequestController.cs [hunk 3]
```diff
@@ -16,5 +23,48 @@ public async Task<LeaveRequestHistoryResponseDto> GetLeaveRequestHistory([FromQu
...
+        public async Task<List<SubordinateLeaveDto>> GetSubordinatesLeaveRequest(int id, LeaveRequestStatus status,CancellationToken cancellationToken)
+        {
+            GetSubordinateLeaveQuery query= new GetSubordinateLeaveQuery();
+            query.Id =id;
+            query.StatusId = status;
+
+            return await _mediator.Send(query,cancellationToken);
+        }
+
+        [HttpPatch("subordinate-leave-requets/{id}")]
```
Comment (zameelnp) [inline src/services/WolfDen.API/Controllers/LeaveManagement/LeaveRequestController.cs:49]: typo

File: src/services/WolfDen.API/Controllers/LeaveManagement/LeaveRequestController.cs [hunk 4]
```diff
@@ -16,5 +23,48 @@ public async Task<LeaveRequestHistoryResponseDto> GetLeaveRequestHistory([FromQu
...
+        public async Task<bool> ApproveOrRejectLeave(int id , [FromBody] ApproveOrRejectLeaveRequestCommand command,CancellationToken cancellationToken)
+        {
+            command.SuperiorId = id;   
+            return await _mediator.Send(command,cancellationToken);
+        }
+
+        [HttpPatch("edit-leave/{id}")]
```
Comment (zameelnp) [inline src/services/WolfDen.API/Controllers/LeaveManagement/LeaveRequestController.cs:56]: use put for updating if there is more than 2 column changes

File: src/services/WolfDen.API/Controllers/LeaveManagement/LeaveRequestController.cs (hunk 2 above)
Comment (mukhil-geeky) [reply src/services/WolfDen.API/Controllers/LeaveManagement/LeaveRequestController.cs:39]: done

File: src/services/WolfDen.API/Controllers/LeaveManagement/LeaveRequestController.cs (hunk 4 above)
Comment (mukhil-geeky) [reply src/services/WolfDen.API/Controllers/LeaveManagement/LeaveRequestController.cs:56]: done

File: src/services/WolfDen.API/Controllers/LeaveManagement/LeaveRequestController.cs (hunk 1 above)
Comment (mukhil-geeky) [reply src/services/WolfDen.API/Controllers/LeaveManagement/LeaveRequestController.cs:33]: Done

Comment (mukhil-geeky) [reply src/services/WolfDen.API/Controllers/LeaveManagement/LeaveRequestController.cs:33]: done

File: src/services/WolfDen.Application/Requests/Commands/LeaveManagement/AddLeaveRequestForEmployeeByAdmin/AddLeaveRequestForEmployeeByAdminHandler.cs [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+            if (!dateCheck)
+            {
+                if (employee.Gender.HasValue && ((employee.Gender == Gender.Male && leaveType.LeaveCategoryId != LeaveCategory.Maternity) || (employee.Gender == Gender.Female && leaveType.LeaveCategoryId != LeaveCategory.Paternity)))
+                {
+                    if (leaveType.LeaveCategoryId == LeaveCategory.WorkFromHome && currentDate < request.FromDate)
+                    {
+                        return await AddLeave();
+                    }
+
+                    else if (currentDate.DayNumber < request.FromDate.DayNumber)
```
[the remark this answers is not in the document]
Comment (mukhil-geeky) [reply src/services/WolfDen.Application/Requests/Commands/LeaveManagement/AddLeaveRequestForEmployeeByAdmin/AddLeaveRequestForEmployeeByAdminHandler.cs:70]: ok

File: src/services/WolfDen.Application/Requests/Commands/LeaveManagement/LeaveRequestDays/AddLeaveRequestDaysCommandHandler.cs [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+        public async Task<bool> Handle(AddLeaveRequestDayCommand request, CancellationToken cancellationToken)
+        {
+            foreach(DateOnly date in request.Date)
+            {
+                LeaveRequestDay leaveRequestDay = new LeaveRequestDay(request.LeaveRequestId,date);
+                _context.Add(leaveRequestDay);
+
+            }
+            int result = await _context.SaveChangesAsync(cancellationToken);
```
[the remark this answers is not in the document]
Comment (mukhil-geeky) [reply src/services/WolfDen.Application/Requests/Commands/LeaveManagement/LeaveRequestDays/AddLeaveRequestDaysCommandHandler.cs:19]: done

File: src/ui/src/app/dashboard/dashboard-body/main/leave-management/subordinate-leave-request/subordinate-leave-request.component.ts [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+      }
+    });
+    
+  }
+
+  reject(id :number){
+    console.log(id);
+    this.approveRejectLeave.leaveRequestId = id;
+    this.approveRejectLeave.statusId = LeaveRequestStatus.Rejected;
+    this.leaveManagementService.approveOrRejectLeave(this.approveRejectLeave).subscribe({
```
[the remark this answers is not in the document]
Comment (mukhil-geeky) [reply src/ui/src/app/dashboard/dashboard-body/main/leave-management/subordinate-leave-request/subordinate-leave-request.component.ts:88]: done

File: src/ui/src/app/dashboard/dashboard-body/main/leave-management/subordinate-leave-request/subordinate-leave-request.component.ts [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+  }
+  
+  ngOnInit(): void {
+    this.loadLeaveRequests(this.open);
+  }
+
+  approve(id :number){
+    this.approveRejectLeave.leaveRequestId = id;
+    this.approveRejectLeave.statusId = LeaveRequestStatus.Approved;
+    this.leaveManagementService.approveOrRejectLeave(this.approveRejectLeave).subscribe({
```
[the remark this answers is not in the document]
Comment (mukhil-geeky) [reply src/ui/src/app/dashboard/dashboard-body/main/leave-management/subordinate-leave-request/subordinate-leave-request.component.ts:68]: done

File: src/ui/src/app/dashboard/dashboard-body/main/leave-management/subordinate-leave-request/subordinate-leave-request.component.ts [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+  };
+  
+  leaveManagementService = inject(LeaveManagementService);
+
+
+
+  loadLeaveRequests(filterStatus : LeaveRequestStatus): void {
+    console.log(filterStatus)
+    this.leavestatus = filterStatus
+    this.leaveManagementService.getSubordinateLeaverequest(this.leavestatus).subscribe({
```
[the remark this answers is not in the document]
Comment (mukhil-geeky) [reply src/ui/src/app/dashboard/dashboard-body/main/leave-management/subordinate-leave-request/subordinate-leave-request.component.ts:35]: done

File: src/ui/src/app/dashboard/dashboard-body/main/leave-management/subordinate-leave-request/subordinate-leave-request.component.ts [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+    leaveRequestId : null,
+    statusId : null,
+  };
+  
+  leaveManagementService = inject(LeaveManagementService);
+
+
+
+  loadLeaveRequests(filterStatus : LeaveRequestStatus): void {
+    console.log(filterStatus)
```
[the remark this answers is not in the document]
Comment (mukhil-geeky) [reply src/ui/src/app/dashboard/dashboard-body/main/leave-management/subordinate-leave-request/subordinate-leave-request.component.ts:33]: done

File: src/services/WolfDen.Application/Requests/Commands/LeaveManagement/AddLeaveRequestForEmployeeByAdmin/AddLeaveRequestForEmployeeByAdminHandler.cs [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+namespace WolfDen.Application.Requests.Commands.LeaveManagement.AddLeaveRequestForEmployeeByAdmin
+{
+    public class AddLeaveRequestForEmployeeByAdminHandler(WolfDenContext context, IMediator mediator) : IRequestHandler<AddLeaveRequestForEmployeeByAdmin, bool>
+    {
+        private readonly WolfDenContext _context= context;
+        private readonly IMediator _mediator = mediator;
+        public async Task<bool> Handle(AddLeaveRequestForEmployeeByAdmin request, CancellationToken cancellationToken)
+        {
+            Employee employee = await _context.Employees.FirstOrDefaultAsync(x => x.EmployeeCode == request.EmployeeCode, cancellationToken);
+            if (employee == null)
```
[the remark this answers is not in the document]
Comment (mukhil-geeky) [reply src/services/WolfDen.Application/Requests/Commands/LeaveManagement/AddLeaveRequestForEmployeeByAdmin/AddLeaveRequestForEmployeeByAdminHandler.cs:20]: done

**Pull request closed without merging**